servers/apps/test/malloc_test.c

```diff
@@ -0,0 +1,30 @@
+#include <resea/malloc.h>
+#include <resea/printf.h>
+#include <string.h>
+#include "test.h"
+#define NUM_PTRS 16
+
+void malloc_test(void) {
+    malloc_init();
+    void *ptr[NUM_PTRS];
+    memset(ptr, 0, sizeof(ptr));
+    int sz = 1;
+
+    for (size_t i = 0; i < NUM_PTRS - 1; i++) {
+        ptr[i] = malloc(sz << i);
+        TEST_ASSERT(ptr[i] != NULL);
+    }
+
+    ptr[NUM_PTRS - 1] = malloc((1 << 15) + 8);
+    for (size_t i = 0; i < NUM_PTRS; i++) {
+        free(ptr[i]);
+    }
+    for (size_t i = 0; i < NUM_PTRS - 1; i++) {
+        ptr[i] = malloc(sz << i);
+        TEST_ASSERT(ptr[i] != NULL);
+    }
+    ptr[NUM_PTRS - 1] = malloc((1 << 15) + 8);
+    for (size_t i = 0; i < NUM_PTRS; i++) {
+        free(ptr[i]);
+    }
```

Comment:
It looks this part does same thing. Please remove it.

Reply:
I did that to see if the same sized chunk is re-allocated. Is that fine?

Reply:
Good point! Let's leave this part as it is but could you add a comment to make your intention explicit?

Reply:
Yes, I think I should have done it previously. Nevertheless, done now.